base_layer/core/src/transactions/transaction_components/encrypted_data.rs

```diff
@@ -29,50 +29,43 @@ use std::mem::size_of;
 
 use borsh::{BorshDeserialize, BorshSerialize};
 use chacha20poly1305::{
-    aead::{Aead, Error, Payload},
+    aead::{AeadCore, AeadInPlace, Error, OsRng},
     KeyInit,
     Tag,
     XChaCha20Poly1305,
     XNonce,
 };
 use digest::{generic_array::GenericArray, FixedOutput};
-use rand::{rngs::OsRng, RngCore};
 use serde::{Deserialize, Serialize};
 use tari_common_types::types::{Commitment, PrivateKey};
-use tari_crypto::{hash::blake2::Blake256, hashing::DomainSeparatedHasher};
+use tari_crypto::{hash::blake2::Blake256, hashing::DomainSeparatedHasher, keys::SecretKey};
 use tari_utilities::{
     hex::{from_hex, to_hex, Hex, HexError},
     safe_array::SafeArray,
     ByteArray,
     ByteArrayError,
 };
 use thiserror::Error;
-use zeroize::Zeroize;
+use zeroize::{Zeroize, Zeroizing};
 
 use super::EncryptedDataKey;
 use crate::transactions::{tari_amount::MicroTari, TransactionSecureNonceKdfDomain};
 
-const VALUE_SIZE: usize = size_of::<u64>(); // 8 bytes
-const KEY_SIZE: usize = size_of::<PrivateKey>(); // 32 bytes
-const SIZE: usize = VALUE_SIZE + KEY_SIZE + size_of::<Tag>() + size_of::<XNonce>(); // 80 bytes
-const BORSH_64: usize = 64;
-const BORSH_X: usize = SIZE - BORSH_64; // 16 bytes
+/// Total encrypted data size: nonce, encrypted value, encrypted mask, tag
+const SIZE: usize = size_of::<XNonce>() + size_of::<u64>() + PrivateKey::KEY_LEN + size_of::<Tag>();
```

Comment:
**Nit**: I would still prefer these and use them throughout as I think they enhance the readability of the code:
```suggestion
const NONCE_SIZE: usize = size_of::<XNonce>();
const VALUE_SIZE: usize = size_of::<u64>();
const KEY_SIZE: usize = PrivateKey::KEY_LEN;
const SIZE: usize = NONCE_SIZE + VALUE_SIZE + KEY_SIZE + size_of::<Tag>();
```

Reply:
Good idea! Done.